syntax = "proto3";

package search;

import "common/track.proto"; // Assuming this is where your Track message is defined

// SearchServiceRequest: The search query input from the user
message SearchServiceRequest {
    string query = 1; // The user input for searching (e.g., track name, artist name)
    int32 limit = 2;  // Optional: Limit of how many results to return
}

// SearchServiceResponse: The response containing a list of tracks that match the query
message SearchServiceResponse {
    repeated common.Track tracks = 1;  // A list of matching tracks
    int32 total_results = 2;           // The total number of search results
}

// SearchService: The gRPC service definition for searching tracks
service SearchService {
    rpc SearchTracks (SearchServiceRequest) returns (SearchServiceResponse);
}